test/sequential/test-fs-readfile.js

```diff
@@ -0,0 +1,72 @@
+'use strict';
+
+const common = require('../common');
+
+const assert = require('assert');
+const fs = require('fs');
+const path = require('path');
+
+// In case openssl is not available
+function slowRandomBytes(len) {
+  const buf = Buffer.alloc(len);
+  for (let i = 0; i < len; i++)
+    buf[i] = Math.random();
+  return buf;
+}
+
+common.refreshTmpDir();
+
+const fileInfo = [ { 'name': path.join(common.tmpDir, '1Kfile.txt'),
+                     'len': 1024,
+},
+                   { 'name': path.join(common.tmpDir, '64Kfile.txt'),
+                     'len': 64 * 1024,
+                   },
+                   { 'name': path.join(common.tmpDir, '1Mfile.txt'),
+                     'len': 1 * 1024 * 1024,
+                   },
+];
+
+/* Populate each fileInfo (and file) with unique fill. */
+for (let i = 0; i < fileInfo.length; i++) {
+  fileInfo[i].fill = slowRandomBytes(fileInfo[i].len);
+  fs.writeFileSync(fileInfo[i].name, fileInfo[i].fill);
+}
+
+/* Test readFile on each size. */
+for (let i = 0; i < fileInfo.length; i++) {
+  fs.readFile(fileInfo[i].name, function(err, buf) {
+    validate(fileInfo[i], err, buf);
+  });
+}
+
+function validate(fileInfo, err, buf) {
+  console.log(`Validating readFile on file of length len ${fileInfo.len}`);
+  assert.ifError(err);
+  assert.notStrictEqual(buf, undefined);
+  assert.notStrictEqual(fileInfo.fill, undefined);
+  assert.ok(fileInfo.fill.equals(buf));
+}
+
+/* Make sure we don't leave test fileInfo in event of failure. */
+function destroy() {
+  for (let i = 0; i < fileInfo.length; i++) {
+    try {
+      fs.unlinkSync(fileInfo[i].name);
+    } catch (err) {
+      // it may not exist
+    }
+  }
+}
+
+process.on('exit', destroy);
+
+process.on('SIGINT', function() {
+  destroy();
+  process.exit();
+});
+
+process.on('uncaughtException', function(err) {
+  destroy();
+  throw err;
+});
```

Comment:
All this cleanup code is not necessary, `common.tmpDir` gets cleaned out automatically.